Wait when trying to play a webstream while connecting to wifi

SZenglein opened the pull request:
The previous code had some issues. First of all, with DONT_ACCEPT_SAME_RFID_TWICE it would not work at all. But without a change, not even that works correctly (due to gTriedToConnectToHost never being set).

IMHO the code is simplified a lot. The whole control flow can be simplified if we just wait until a connection try hast completed.

The behavior will be slightly different. Most importantly, loading a webstream as the last played rfid now works consistently. Further, the error handling of playing a webstream while disconnected is shifted to the audio library and is handled like any other error.

pitlochry commented:
Super Idee. Danke fürs Reinstellen. Ich hoffe, es ist ok, wenn ich hier deutsch schreibe. Ich habe deinen Code noch nicht ausprobiert, beim Lesen sind mir jedoch folgende Gedanken gekommen:

- wie lange probiert der espuino eine wlan connection zu bekommen? Wenn er dies zu lange versucht, sollte man ggf. hier noch ein maximales wifiConnectionWaitInterval einführen, dass nicht zu lange gewartet wird, z.B. bei schlechtem oder keinem wlan oder sonstigen wlan Problemen. Über connectStartTimestamp bekommt man einen timestamp, wann mit dem Verbindungsversuch begonnen wurde, das könnte hilfreich sein (?)

- wenn während des Wartens auf eine Wlan Verbindung eine andere MusicRfid aufgelegt wird (insbesondere kein webstream), dann sollte doch sinnvollerweise die Warteschleife durchbrochen werden, oder? 

- ggf. wäre noch ein error-log sinnvoll, warum der webstream nicht abgespielt werden kann, z.B. 
o "wifi is enabled, but not connected, no webstream possible" (warum auch immer) 
o "wifi is not enabled, no webstream possible"

Keine Ahnung, ob das wichtig ist, waren nur meine Gedanken.

Commit:
Fix playing a webstream while still connecting to WiFi

The code is simplified a lot. I removed the functions and variables that
were never running anyway.

The webstream command is now always accepted, and the error is handled
inside the audio task. The audio task also blocks as long as a
connection try is in progress.

In addition to the code that did not work anyway, this should also fix
the oversight that we want to immediately play a different webstream
after starting, not only the last RFID.
